core/runtime/impl/grandpa_impl.cpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "runtime/impl/grandpa_impl.hpp"
+
+#include "runtime/impl/wasm_memory_stream.hpp"
+#include "scale/collection.hpp"
+#include "scale/optional.hpp"
+#include "scale/scale_error.hpp"
+
+namespace kagome::runtime {
+  using primitives::Digest;
+  using primitives::ForcedChangeType;
+  using primitives::ScheduledChange;
+  using primitives::SessionKey;
+  using primitives::WeightedAuthority;
+
+  GrandpaImpl::GrandpaImpl(common::Buffer state_code,
+                           std::shared_ptr<extensions::Extension> extension,
+                           std::shared_ptr<primitives::ScaleCodec> codec)
+      : memory_(extension->memory()),
+        codec_(std::move(codec)),
+        executor_(std::move(extension)),
+        state_code_(std::move(state_code)) {}
+
+  outcome::result<std::optional<ScheduledChange>> GrandpaImpl::pendingChange(
+      const Grandpa::Digest &digest) {
+    OUTCOME_TRY(encoded_digest, codec_->encodeDigest(digest));
+
+    runtime::SizeType ext_size = encoded_digest.size();
+    runtime::WasmPointer ptr = memory_->allocate(ext_size);
+    memory_->storeBuffer(ptr, encoded_digest);
+
+    wasm::LiteralList ll{wasm::Literal(ptr), wasm::Literal(ext_size)};
+
+    OUTCOME_TRY(
+        res,
+        executor_.call(state_code_, "GrandpaApi_grandpa_pending_change", ll));
+
+    uint32_t res_addr = static_cast<uint64_t>(res.geti64()) & 0xFFFFFFFFull;
```

Comment:
And in my PR I added a function that does this operation, so perhaps you could either wait for its merge and then update this line or at least leave a TODO here to keep runtime api implementations consistent.

Reply:
fixed